### PR TITLE
[MM-48077] - Notify admin is received for @unknown user

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Because profiles are fetched in batches/pages, it's possible to have missing profiles in the current page for workspaces with a lot of users. This PR adds changes to fetch the missing profiles.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48077

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by nickmisasi — APPROVED** [review]
Really nice you don't have to deal with the pagination!

File: components/custom_open_pricing_modal_post_renderer/index.tsx [hunk 1]
```diff
@@ -71,13 +74,22 @@ export default function OpenPricingModalPost(props: {post: Post}) {
...
     const requestFeatures = postProps?.requested_features;
     const wasTrialRequest = postProps?.trial;
+
+    useEffect(() => {
+        if (requestFeatures) {
+            for (const featureId of Object.keys(requestFeatures)) {
+                dispatch(getMissingProfilesByIds(getUserIdsForUsersThatRequestedFeature(requestFeatures[featureId])));
+            }
+        }
+    }, []);
```
Comment (marianunez) [inline components/custom_open_pricing_modal_post_renderer/index.tsx:88]: Do we need to add a condition to not always run this on every render?
```suggestion
    useEffect(() => {
        if (requestFeatures) {
            for (const featureId of Object.keys(requestFeatures)) {
                dispatch(getMissingProfilesByIds(getUserIdsForUsersThatRequestedFeature(requestFeatures[featureId])));
            }
        }
    }, [requestFeatures]);
```

Comment (AGMETEOR) [reply components/custom_open_pricing_modal_post_renderer/index.tsx]: `[]` as second arg ensures it runs once

Comment (marianunez) [reply components/custom_open_pricing_modal_post_renderer/index.tsx]: Yes, you're right, we don't need to add anything else then

**Review by marianunez — APPROVED** [review]

**Comment by AGMETEOR** [comment]
@yasserfaraazkhan this might be a bit difficult to test. First we need to load the server with a bunch of users >100 and then make notifications for the 100 + n user

**Comment by marianunez** [comment]
> @yasserfaraazkhan this might be a bit difficult to test. First we need to load the server with a bunch of users >100 and then make notifications for the 100 + n user

It should be possible to load the server with that many users with the `/test` command, is that correct @stevemudie ? Maybe you can provide some details on which command would work here

**Comment by stevemudie** [comment]
@marianunez @yasserfaraazkhan 
> It should be possible to load the server with that many users with the `/test` command, is that correct @stevemudie ? Maybe you can provide some details on which command would work here

You can add as many users as you'd like with `/test users [fuzz] <Min Users> <Max Users>` but I'm not aware of any way to then use those users to send admin requests. Assuming that's what @AGMETEOR is referring to with "_…then make notifications for the 100 + n user_"?

**Comment by marianunez** [comment]
> "_…then make notifications for the 100 + n user_"?

I guess this means that you need to log in as user 101 and perform a Notify Admin request.

**Comment by yasserfaraazkhan** [comment]
> @yasserfaraazkhan this might be a bit difficult to test. First we need to load the server with a bunch of users >100 and then make notifications for the 100 + n user

Sure @AGMETEOR @Steve, I'll try out.

**Comment by yasserfaraazkhan** [comment]
Hi @AGMETEOR , @stevemudie ,
I was getting this notification while trying /test cmd 
Screenshot 2022-11-03 at 12 07 09 PM


I've used cypress to generate Notifications from 100 users to upgrade, this is how the screen appeared.

Screenshot 2022-11-03 at 11 33 18 AM

in the list there were few users with '@unknow' tag.
Screenshot 2022-11-03 at 12 37 21 PM

I'll try on PR env too.

**Comment by AGMETEOR** [comment]
> Hi @AGMETEOR , @stevemudie , I was getting this notification while trying /test cmd Screenshot 2022-11-03 at 12 07 09 PM
> 
> I've used cypress to generate Notifications from 100 users to upgrade, this is how the screen appeared. Screenshot 2022-11-03 at 12 42 42 PM
> 
> in the list there were few users with '@unknow' tag. Screenshot 2022-11-03 at 12 37 21 PM
> 
> I'll try on PR env too.

@yasserfaraazkhan the same fix has been applied to the modal. Try and test again with cypress

**Comment by marianunez** [comment]
@amyblais heads up we are cherry-picking this fix to `cloud`

**Pull request merged**